src/main.py

```diff
@@ -233,26 +227,47 @@ def internet_on():
 		return False
 
 
-def gettoken():
-	token = mc.get("access_token")
-	refresh = config['alexa']['refresh_token']
-	if token:
-		return token
-	elif refresh:
+class Token(object):
+
+	_token = None
+	_timestamp = None
+	_validity = 3570
+
+	def __init__(self, aconfig):
+
+		self._aconfig = aconfig
+
+		self.renew()
+
+	def get(self):
+
+		if time.time() - self._timestamp > self._validity:
+			self.renew()
+
+		return self._token
+
+	def renew(self):
+
+		logger.debug("AVS token: Requesting a new one")
+
+		if not self._aconfig['refresh_token']:
+			raise ValueError("refresh_token is not valid")
+
 		payload = {
-			"client_id": config['alexa']['Client_ID'],
-			"client_secret": config['alexa']['Client_Secret'],
-			"refresh_token": refresh,
+			"client_id": self._aconfig['Client_ID'],
+			"client_secret": self._aconfig['Client_Secret'],
+			"refresh_token": self._aconfig['refresh_token'],
 			"grant_type": "refresh_token"
 		}
+
 		url = "https://api.amazon.com/auth/o2/token"
 		response = requests.post(url, data=payload)
 		resp = json.loads(response.text)
-		mc.set("access_token", resp['access_token'], 3570)
-		return resp['access_token']
-	else:
-		return False
 
+		self._token = resp['access_token']
+		self._timestamp = time.time()
+
+		logger.debug("AVS token: Obtained successfully")
```

Comment:
This could be `info` too - it is good to know that we've asked for token and obtained it.